advanced_filters/static/advanced-filters/advanced-filters.js

```diff
@@ -70,7 +70,11 @@ var OperatorHandlers = function($) {
 						  MODEL_LABEL) + '/' + field;
 		var input = $(elm).parents('tr').find('input.query-value');
 		input.select2("destroy");
+		var value = input.val();
 		$.get(choices_url, function(data) {
+			if (value) {
+				data.results.push({'id': value, 'text': value})
+			}
 			input.select2({'data': data, 'createSearchChoice': function(term) {
                 return { 'id': term, 'text': term };
```

Comment:
Perhaps we could simply [call `input.val(value)`"](https://select2.github.io/options.html#what-events-does-select2-listen-for) to select the pre-initialized value as current.